pipeline/ffmpeg.go

```diff
@@ -31,15 +37,43 @@ func (f *ffmpeg) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 	job.SegmentingTargetURL = segmentingTargetURL.String()
 	log.AddContext(job.RequestID, "segmented_url", job.SegmentingTargetURL)
 
-	// TODO: Segment
+	// Begin Segmenting
 	log.Log(job.RequestID, "Beginning segmenting via FFMPEG/Livepeer pipeline")
 	job.ReportProgress(clients.TranscodeStatusPreparing, 0.5)
 
-	// TODO: Segmenting Finished
+	// FFMPEG fails when presented with a raw IP + Path type URL, so we prepend "http://" to it
+	internalAddress := config.HTTPInternalAddress
+	if !strings.HasPrefix(internalAddress, "http") {
+		internalAddress = "http://" + internalAddress
+	}
+
+	cmd := exec.Command(
+		"ffmpeg",
+		"-i", job.SignedSourceURL, // Input MP4 / MOV file
+		"-c", "copy", // Make sure we don't try to do any transcoding
+		"-f", "hls", // Output to HLS (segments + manifest)
+		"-hls_segment_type", "mpegts", // Use mpegts rather than fmp4 segments for compatibility reasons
+		"-hls_playlist_type", "vod", // Mark this as a VOD rather than Live playlist
+		"-hls_list_size", "0", // Unbounded playlist size since this isn't Live
+		"-hls_time", strconv.FormatInt(job.TargetSegmentSizeSecs, 10), // Tell FFMPEG what segment size (secs) to aim for
+		"-method", "PUT", // Send the segments and playlist to ourselves over HTTP as they're ready
+		fmt.Sprintf("%s/api/ffmpeg/%s/index.m3u8", internalAddress, job.StreamName), // Local HTTP API to send to
+	)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
```

Comment:
Agree, and we already have this library in catalyst-api. I rolled it in for the mp4 generation stuff. See this for an example: 
https://github.com/livepeer/catalyst-api/blob/main/video/transmux.go

Also, all the video processing logic should go under video/segment.go or something like that so that we have separation between video processing bits and everything else.

Reply:
Yeah, agree with both your points - it was a bit faster for me to prototype with the raw Exec stuff but now's probably a good time to make the switch

Reply:
Done!